### ISSUE TITLE
Speech "cuts off" at the end of each sentence

### ISSUE THREAD

**Issue by Thelukepet** [ysharma3501/LuxTTS#23]
Hey, I tried reducing the speed for the `.generate_speech` and increasing the `duration` of the '.encode_prompt` but it did not fix the issue when using english samples + ref_text.

In the end I fixed it by changing the `generate` function in the `modeling_utils.py`, specifically by adding padding to the Vocoder to give it enough data to finish the previous sound. Anything else I should have looked at?
```
def generate(
    prompt_tokens,
    prompt_features_lens,
    prompt_features,
    prompt_rms,
    text,
    model,
    vocoder,
    tokenizer,
    num_step=4,
    guidance_scale=3.0,
    speed=1.0,
    t_shift=0.5,
    target_rms=0.1,
):
    tokens = tokenizer.texts_to_token_ids([text])
    device = next(model.parameters()).device  # Auto-detect device

    speed = speed * 1.3

    with torch.inference_mode():
        (pred_features, _, _, _) = model.sample(
            tokens=tokens,
            prompt_tokens=prompt_tokens,
            prompt_features=prompt_features,
            prompt_features_lens=prompt_features_lens,
            speed=speed,
            t_shift=t_shift,
            duration="predict",
            num_step=num_step,
            guidance_scale=guidance_scale,
        )

    # Convert to waveform
    pred_features = pred_features.permute(0, 2, 1) / 0.1

    # FIX: Padding for the Vocoder
    # We take the last frame and repeat it 15 times (approx 150ms buffer)
    # This gives Vocos enough data to finish the previous sound without cutting it.
    last_frame = pred_features[:, :, -1:]
    padding_frames = last_frame.repeat(1, 1, 15)
    pred_features = torch.cat([pred_features, padding_frames], dim=2)

    wav = vocoder.decode(pred_features).squeeze(1).clamp(-1, 1)

    # Volume matching
    if prompt_rms < target_rms:
        wav = wav * (prompt_rms / target_rms)

    return wav
```

**Comment by MichaelBTryin** [comment]
It took me so long tinkering and tweaking settings, and implementing features to try and solve this. I think I finally got it solved for me. All I can do is share my configs, and you'll gave to implement "lead_in_silence" and FadeOut Features. Although OP, you seemed to have solved it, so that's awesome!

{
  "TTS": {
    "normalize_audio": true,
    "normalization_headroom": 0.9,
    "gain_db": 0.0,
    "fade_duration": 0.03,
    "fadeout_target": 0.0,
    "silence_duration": 0.2,
    "lead_in_silence": 0.3,
    "post_silence_duration": 0.05,
    "_comment": "Configs not directly apart of LuxTTS, but rather features I added on in attempt to make it function well, you can seek to implement features like this as needed/wanted"
  },
  "luxtts": {
    "model_id": "YatharthS/LuxTTS",
    "device": "cuda:0",
    "threads": 4,
    "ref_duration": 15,
    "rms": 0.01,
    "num_steps": 8,
    "t_shift": 0.9,
    "return_smooth": false,
    "speed": 0.75,
    "guidance_scale": 1.9,
    "max_threads": 8,
    "max_num_steps": 12,
    "comment": "Core LuxTTS Configs"
  }
}
